packages/vite/src/node/config.ts

```diff
@@ -960,38 +960,44 @@ async function bundleConfigFile(
       {
         name: 'externalize-deps',
         setup(build) {
-          const options: InternalResolveOptions = {
-            root: path.dirname(fileName),
-            isBuild: true,
-            isProduction: true,
-            isRequire: !isESM,
-            preferRelative: false,
-            tryIndex: true,
-            mainFields: [],
-            browserField: false,
-            conditions: [],
-            dedupe: [],
-            extensions: DEFAULT_EXTENSIONS,
-            preserveSymlinks: false
-          }
-
-          build.onResolve({ filter: /.*/ }, ({ path: id, importer, kind }) => {
-            // externalize bare imports
-            if (id[0] !== '.' && !path.isAbsolute(id) && !isBuiltin(id)) {
-              // partial deno support as `npm:` does not work in `tryNodeResolve`
+          // externalize bare imports
+          build.onResolve(
+            { filter: /^[^.].*/ },
+            async ({ path: id, importer, kind, resolveDir, pluginData }) => {
+              if (pluginData?.internalResolve) {
+                return
+              }
+              if (kind === 'entry-point' || path.isAbsolute(id)) {
+                return
+              }
+
+              // partial deno support as `npm:` does not work with esbuild
               if (id.startsWith('npm:')) {
                 return { external: true }
               }
-              let idFsPath = tryNodeResolve(id, importer, options, false)?.id
-              if (idFsPath && (isESM || kind === 'dynamic-import')) {
-                idFsPath = pathToFileURL(idFsPath).href
+              const result = await build.resolve(id, {
+                importer,
+                kind:
+                  kind === 'import-statement' && !isESM ? 'require-call' : kind,
+                resolveDir,
+                pluginData: { internalResolve: true }
+              })
+              if (result.errors.length > 0) {
+                return { errors: result.errors }
+              }
+              if (result.external) {
+                return { path: result.path, external: true }
               }
+              const absolutePath = path.resolve(importer, '..', result.path)
```

Comment:
```suggestion
              const absolutePath = path.resolve(path.dirname(importer), result.path)
```

I'm not sure if it makes a difference, but I usually use `path.dirname`. But the `..` trick here seems to work too.